docs: Add section on building rocprofsys for no GPU

peterjunpark opened the pull request:
Adds brief information in `install.rst` regarding
- Provided build scripts
- What to do when building rocprofiler-systems w/o ROCm
- Recent changes to the configuration settings as a result of rocprofiler-sdk support.

Fixes minor formatting issues in `install.rst` resulting in Sphinx build warnings

peterjunpark committed:
add section building rocprof-sys w/o GPU

peterjunpark commented:
preview: https://advanced-micro-devices-demo--57.com.readthedocs.build/projects/rocprofiler-systems/en/57/install/install.html#rocm-systems-profiler-without-rocm

peterjunpark commented:
@pranswarup could you take a look?

dgaliffiAMD commented:
Another option would be to point users to the provided build scripts: 

```
./scripts/build-release.sh --help
Options:
    --core       [+nopython] [+python]                    Core (Use '+nopython' to build w/o python, use '+python' to python build with python)
    --mpi        [+nopython] [+python]                    MPI (Use '+nopython' to build w/o python, use '+python' to python build with python)
    --rocm       [+nopython] [+python]                    ROCm (Use '+nopython' to build w/o python, use '+python' to python build with python)
    --rocm-mpi   [+nopython] [+python]                    ROCm + MPI (Use '+nopython' to build w/o python, use '+python' to python build with python)
    --mpi-impl   [openmpi|mpich]                          MPI implementation

    --lto                  [on|off]                       Enable LTO (default: off)
    --strip                [on|off]                       Strip libraries (default: off)
    --perfetto-tools       [on|off]                       Install perfetto tools (default: on)
    --static-libgcc        [on|off]                       Build with static libgcc (default: on)
    --static-libstdcxx     [on|off]                       Build with static libstdc++ (default: on)
    --hidden-visibility    [on|off]                       Build with hidden visibility (default: on)
    --max-threads          N                              Max number of threads supported (default: 2048)
    --parallel             N                              Number of parallel build jobs (default: 12)
    --generators           [STGZ][DEB][RPM][+others]      CPack generators (default: stgz deb rpm)
```

So, to build without ROCm-support and create a STGZ installer, you would use `./script/build-release.sh --core +python --generators STGZ`.

peterjunpark committed:
add info on using build scripts

wording

peterjunpark commented:
Added https://advanced-micro-devices-demo--57.com.readthedocs.build/projects/rocprofiler-systems/en/57/install/install.html#using-the-build-script and updated https://advanced-micro-devices-demo--57.com.readthedocs.build/projects/rocprofiler-systems/en/57/install/install.html#rocm-systems-profiler-without-rocm

Could you help make sure it's accurate and provide feedback? @dgaliffiAMD @pranswarup Thanks.

peterjunpark committed:
fix rocm pkg names

pranswarup reviewed (dismissed):
Documentation leverages existing build processes and it only adds information on build flags that can be used to build rocprof-sys without rocm or gpu installed.

dgaliffiAMD commented on docs/install/install.rst:
```diff
@@ -176,9 +175,9 @@ Optional third-party packages
-  * Roctracer for HIP API and kernel tracing
-  * ROCM-SMI for GPU monitoring
-  * Rocprofiler for GPU hardware counters
+  * ROCTracer for HIP API and kernel tracing
+  * ROCm SMI Lib for GPU monitoring
+  * ROCProfiler for GPU hardware counters
```
```suggestion
  * ROCm SMI Lib for GPU monitoring
  * ROCprofiler SDK for GPU hardware counters and ROCm tracing
```
Sorry, updating to reflect some recent changes for the ROCprofiler SDK support.

peterjunpark replied:
Thanks @dgaliffiAMD

dgaliffiAMD commented on docs/install/install.rst:
```diff
@@ -306,6 +338,20 @@ ROCm Systems Profiler on an application built against OpenMPI causes a segmentat
+ROCm Systems Profiler without ROCm
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To build ROCm Systems Profiler for use on systems without a GPU or the ROCm runtime, disable HIP
+support using the CMake configuration option ``ROCPROFSYS_USE_HIP=OFF``. See :ref:`cmake-options`
```
```suggestion
To build ROCm Systems Profiler for use on systems without a GPU or the ROCm runtime, disable ROCm
support using the CMake configuration option ``ROCPROFSYS_USE_ROCM=OFF``. See :ref:`cmake-options`
```
Sorry, updating to reflect some recent changes to the configuration settings with the rocprofiler-sdk support.

peterjunpark replied:
Thanks

peterjunpark replied:
ref: https://github.com/ROCm/rocprofiler-systems/pull/55

peterjunpark committed:
Apply suggestions from code review

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

peterjunpark commented:
@ROCm/rocm-documentation

amd-jnovotny reviewed (approved):
looks good.
